### PR TITLE
Feat : Added Parametrized Routing

### PR DESCRIPTION
**Pull request by srejitk**
The following Feature request contains 

1. Parametrized Routing
2. Conversion to modular CSS
3. API Function Call Fixes
4. Responsiveness
5. Formatting

### CODE REVIEW

File: House-of-glass/src/Contexts/Category/CategoryContext.jsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+const CategoryProvider = ({ children }) => {
+  const [categoryList, setCategoryList] = useState([]);
+  const navigate = useNavigate();
+
+  const fetchData = async () => {
+    try {
+      const { data, status } = await axios.get("/api/categories");
+      if (status === 200) {
+        setCategoryList(() => data?.categories);
+        console.log("Cats are here");
```
Comment (SimranjitKaurMaan) [inline House-of-glass/src/Contexts/Category/CategoryContext.jsx:20]: Remove console.log.

File: House-of-glass/src/components/Filters/Filters.jsx [hunk 2]
```diff
@@ -210,13 +249,108 @@ export default function Filters() {
...
+        `}
+        >
+          <span className="material-icons">production_quantity_limits</span>
+          <label htmlFor="category-1">Ray Ban</label>
+          <input
+            type="checkbox"
+            name="CATEGORY"
+            value="RAYBAN"
+            checked={
+              productState.categoryState.find(
```
Comment (SimranjitKaurMaan) [inline House-of-glass/src/components/Filters/Filters.jsx:303]: Can be abstracted out into a function.

**Review by SimranjitKaurMaan — APPROVED** [review]
Minor suggestions. Rest looks good.

File: House-of-glass/src/Contexts/Category/CategoryContext.jsx (hunk 1 above)
Comment (coldpigli) [inline House-of-glass/src/Contexts/Category/CategoryContext.jsx:20]: Remove logs

File: House-of-glass/src/components/Filters/Filters.jsx [hunk 3]
```diff
@@ -210,13 +249,108 @@ export default function Filters() {
...
+        `}
+        >
+          <span className="material-icons">production_quantity_limits</span>
+          <label htmlFor="category-1">Vincent Chase</label>
+          <input
+            type="checkbox"
+            name="CATEGORY"
+            value="VINCENTCHASE"
+            checked={
+              productState.categoryState.find(
```
Comment (coldpigli) [inline House-of-glass/src/components/Filters/Filters.jsx:333]: Can extract this out into a function

**Review by coldpigli — APPROVED** [review]
Have suggested minor improvements. Other things looks good.

File: House-of-glass/src/Contexts/Category/CategoryContext.jsx (hunk 1 above)
Comment (srejitk) [reply House-of-glass/src/Contexts/Category/CategoryContext.jsx:20]: Done.

File: House-of-glass/src/components/Filters/Filters.jsx (hunk 2 above)
Comment (srejitk) [reply House-of-glass/src/components/Filters/Filters.jsx:303]: Will do that. Thank You.

File: House-of-glass/src/Contexts/Category/CategoryContext.jsx (hunk 1 above)
Comment (srejitk) [reply House-of-glass/src/Contexts/Category/CategoryContext.jsx:20]: Done.

**Pull request merged**